### ISSUE TITLE
Fix initialization of `google-generativeai`

### ISSUE THREAD

**Issue by NP4567-dev** [genai-impact/ecologits#76]
### Problem:
Starting from a new venv, Ecologits.init() raises the following error when initiating google_instrumentor:
> ModuleNotFoundError: No module named 'google'

Relevant information:
- the project relies on a pyproject.toml but does not use poetry
- google generative-ai is not in the pyproject.toml

### Workaround:
Adding google-generativeai to my pyproject.toml solves the problem.
This works but is not perfect as I install a bunch of packages I'll never need.

### "Easy fix":
- Editing init_google_instrumentor: 
 ` if importlib.util.find_spec("google.generativeai") is not None:`
 becomes:
 ` if importlib.util.find_spec("google-generativeai") is not None:`
Everything works fine in my case but I have no way to validate it works for an user using google-generativeai.

### New feature solution:
Enabling the user to choose which publisher to load at the init.
This could avert the occurrence of an environment-specific issue that renders the entire package unusable.
I started adding such a feature on [my fork](https://github.com/NP4567-dev/ecologits/tree/feature/publisher_config), let me know if you are interested on adding it to the main repo.

**Comment by samuelrince** [comment]
Hey @NP4567-dev, are you using the latest version of EcoLogits (0.3.2) we have already fixed the issue with google in #62 ?

Otherwise, I agree on the fact that we need to let the user choose which tracer to initialize, discussed in #65.

**Comment by NP4567-dev** [comment]
I'm on version 0.3.2, I think the problem comes from having google.generativeai instead of google-generativeai

**Comment by samuelrince** [comment]
Changing `google.generativeai` in the `importlib.util.find_spec` function by `google-generativeai` will make google provider unusable. The check is about finding whether a python module exists or not. For `google-generativeai` library, we need to look for the associated module, which is `google.generativeai`. A module cannot contain dashes "-".

Can you @NP4567-dev, please provide the full error trace and the output of the `pip freeze` command? I don't know what is going on with Google libraries, but we do not have this type of issue with others...

Thanks 🙏

**Comment by samuelrince** [comment]
Thanks, so probably changing to the following would fix that issue. First look for `google`, then `google.generativeai`.

```python
def init_google_instrumentor() -> None:
    if importlib.util.find_spec("google") is not None \
            and importlib.util.find_spec("google.generativeai") is not None:
        from ecologits.tracers.google_tracer import GoogleInstrumentor

        instrumentor = GoogleInstrumentor()
        instrumentor.instrument()
```

**Comment by NP4567-dev** [comment]
I applied this change locally and it works, this seems like a viable fix 🥳 
Thanks for checking 😃

**Comment by samuelrince** [comment]
Fixed by #77
